src/shady-css/parser.js

```diff
@@ -225,22 +229,21 @@ class Parser {
       }
     }
 
-    rule = tokenizer.slice(ruleStart, ruleEnd);
-
     // A ruleset never contains or ends with a semi-colon.
     if (tokenizer.currentToken.is(Token.type.propertyBoundary)) {
-      let colonIndex = rule.indexOf(':');
+      let declarationName = tokenizer.slice(ruleStart, colon.previous);
```

Comment:
As far as the name spanning multiple tokens is concerned: I'm unsure. It may come to pass that someone will conceive of a fancy syntax that would cause this. However, I don't think there is valid CSS that you can write today which would cause this.